File: YesWeHack Dojo Challenge/#40 - Hacker Profile/setup_code.js
const ejs = require("ejs")
const fs = require("fs")
process.chdir('/tmp')

// Add flag as a system enviroment variable
process.env.FLAG = flag

var profiles = [
    {
        "picture": "https://static.vecteezy.com/ti/gratis-vektor/p1/9521808-vintage-eleganta-lejonkungen-krona-illustrationer-vector.jpg",
        "username": "minilucker",
        "isActive": true,
        "description": "a zero-click 0-day is an error that is so violent that you have 0 days to patch it",
        "lastViewed": Date(),
        "impact": 35,
        "points": 1337,
        "bugCount": 102,
    },
    {
        "picture": "https://static.vecteezy.com/ti/gratis-vektor/p1/49009497-sot-socker-segelflygplan-tecknad-serie-illustration-vector.jpg",
        "username": "pwnwithlove",
        "isActive": true,
        "description": "i <3 sugar gliders",
        "lastViewed": Date(),
        "impact": 34,
        "points": 4212,
        "bugCount": 431,
    },
    {
        "picture": "https://static.vecteezy.com/ti/gratis-vektor/p2/47321447-elefant-hog-kvalitet-logotyp-illustration-idealisk-for-t-shirt-grafisk-vector.jpg",
        "username": "BitK",
        "isActive": true,
        "description": "Babar Is The King",
        "lastViewed": Date(),
        "impact": 40,
        "points": 1352,
        "bugCount": 125,
    },
    {
        "picture": "https://static.vecteezy.com/ti/gratis-vektor/p1/24681876-varg-ansikte-logotyp-vektor-illustration-vector.jpg",
        "username": "Brumens",
        "isActive": true,
        "description": "Your beloved Dojo chall maker",
        "lastViewed": Date(),
        "impact": 29,
        "points": 2245,
        "bugCount": 212,
    }
]

// Write the ejs template file
fs.writeFileSync('index.ejs', `
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
</head>    
<body>
    <div class="wrapper">
        <div class="profile-wrapper">
        <% if (error === undefined) { %>
            <div class="profile">
                <img class="icon" src="https://api.iconify.design/basil:upload-solid.svg?color=%23fff">
                <img  class="pic" src="<%= user.picture %>">
                <img class="icon" src="https://api.iconify.design/material-symbols:edit-document.svg?color=%23fff">
            </div>
            <div class="info">
                <div class="field">
                    <div>
                        <img src="https://api.iconify.design/material-symbols-light:swords-rounded.svg?color=%23fff">
                        <b>Impact</b>
                    </div>
                    <p><%= user.impact %></p>
                </div>
                <div class="field">
                    <div>
                        <img src="https://api.iconify.design/bx:stats.svg?color=%23fff">
                        <b>Points</b>
                    </div>
                    <p><%= user.points %></p>
                </div>
                <div class="field">
                    <div>
                        <img src="https://api.iconify.design/solar:bug-bold.svg?color=%23fff">
                        <b>Bugs</b>
                    </div>
                    <p><%= user.bugCount %></p>
                </div>
            </div>

            <div class="field-text">
                <span class="field-key">Username:</span>
                <span class="field-value"><%= user.username %></span>
            </div>
            <div class="field-text">
                <span class="field-key">IsActive:</span>
                <span class="field-value"><%= user.isActive %></span>
            </div>
            <div class="field-text">
                <span class="field-key">Description:</span>
                <span class="field-value"><%= user.description %></span>
            </div>
            <div class="field-text">
                <span class="field-key">Viewed:</span>
                <span class="field-value"><%= user.lastViewed %></span>
            </div>
        <% } else { %>
            <pre id="errorDisplay"><%= error %></pre>
            <% if (logs !== undefined) { %> 
                <hr>
                    <pre id="debugLogs">
                        <%= logs %>
                    </pre>
                <hr>
            <% } %>
        <% } %>
        </div>
    </div>
<style>
@import url('https://fonts.googleapis.com/css2?family=Averia+Sans+Libre:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&family=Changa:wght@200..800&family=Coiny&family=Finger+Paint&family=Jersey+15&family=Knewave&family=Schoolbell&family=Sour+Gummy:ital,wght@0,100..900;1,100..900&display=swap');

:root {
    --color-txt: white;
    --color-bg: #101e11;
    --color-border: #4a4a4a;
    --color-bg-info-field: #f66158;
    --color-bg-profile: #273036;
    --color-bg-field-text: #273036;
}

body {
    background-image: url("https://static.vecteezy.com/ti/gratis-vektor/p1/45711034-monster-med-kvadrater-digital-bla-teknologi-bakgrund-vector.jpg"); // We have licensed account for this
    background-size: cover;
    font-family: "Coiny", serif;
    font-weight: 400;
    font-style: normal;
    background-color: var(--color-bg);
    background-size: cover;
    padding: 0;
    margin: 0;
    color: var(--color-txt);
    font-size: 18px;
}

.wrapper {
    position: fixed;
    display: flex; 
    flex-direction: column; 
    justify-content: space-between; 
    align-items: stretch; 
    border-radius: 25px;
    padding: 20px;
    transform: translate(-50%, -50%);
    top: 50%;
    left: 50%;
}

/* Profile container */
.profile-wrapper {
    background-color: var(--color-bg-profile);
    border: 1px solid var(--color-border);
    border-radius: 22px;
    padding: 30px;
    width: 300px;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
    
}
.profile-wrapper .profile {
    display: flex;
    align-items: center;
}

.profile-wrapper .profile .icon {
    width: 32px;
    height: 32px;
    cursor: pointer;
}

/* Profile picture */
.profile-wrapper .profile .pic {
    width: 110px;
    height: 110px;
    border-radius: 50%;
    margin: 0 auto 20px;
    display: block;
    border: 2px solid var(--color-border);
    object-fit: cover;
}

/* Info fields */
.info {
    padding: 0;
    margin: 0;
    display: inline-flex;
    gap: 12px;
    padding-bottom: 14px;
}
.info .field {
    padding: 6px;
    background-color: var(--color-bg-info-field);
    border-radius: 13px;
    display: grid;
    justify-content: center;
    align-items: center;
    width: 80px;
    height: 50px;
    transition: 0.33s;
    font-size: 14px;
}
.info .field div {
    display: flex;
    gap: 4px;
}
.info .field img {
    width: 24px;
    height: 24px;
}
.info .field p {
    margin: 0;
}
.field:hover {
    cursor: pointer;    
    transform: translate(0, -8px);
}

/* Text field */
.field-text {
    display: flex;
    justify-content: space-between;
    background-color: var(--color-bg-field-text);
    border: 1px solid var(--color-border);
    border-radius: 12px;
    color: #e0e0e0;
    font-size: 16px;
    margin-bottom: 15px;
    padding: 10px;
    transition: background-color 0.3s ease;
}

.field-text:hover {
    background-color: #404040;
}

/* Key (left part) */
.field-key {
    font-weight: bold;
    color: #8a8a8a;
}

/* Value (right part) */
.field-value {
    text-align: right;
    flex-grow: 1;
    margin-left: 10px;
}

/* Editable style for the value */
.field-value[contenteditable="true"]:focus {
    outline: none;
    border-bottom: 1px dashed #6a6a6a;
}

#errorDisplay {
    background-color: #ff4444;
    color: white;
    padding: 10px;
    margin-bottom: 20px;
    border-radius: 5px;
}

#debugLogs {
    background-color: #2a2a2a;
    border: 1px solid var(--color-border);
    border-radius: 5px;
    overflow: auto;
    max-height: 400px;
}

pre {
    margin: 0;
    padding: 10px;
    white-space: pre-wrap;
    word-wrap: break-word;
    line-height: 1.5;
}

.log-entry {
    margin-bottom: 5px;
    border-bottom: 1px solid #3a3a3a;
    padding-bottom: 5px;
}
    </style>
</body>
</html>
`)

return {fs, ejs, profiles}

File: YesWeHack Dojo Challenges/42 - Hex Color Palette/setup_code.py
import os

os.chdir("/tmp/")
os.makedirs("templates", exist_ok=True)
os.makedirs("xml", exist_ok=True)

with open("flag.txt", "w") as f:
    f.write(flag)

with open("xml/sample.xml", 'w') as f:
    f.write('''
<!DOCTYPE colors [
    <!ELEMENT colors (color*)>
    <!ELEMENT color (#PCDATA)>
]>
<colors>
    <color>#FF5733</color>
    <color>#1E3A8A</color>
    <color>#2ECC71</color>
    <color>#F1C40F</color>
    <color>#8E44AD</color>
    <color>#2C3E50</color>
    <color>#FFC0CB</color>
    <color>#00FFFF</color>
</colors>
''')

with open("xml/config.dtd", 'w') as f:
    f.write('''
<!ENTITY % dtd "<!ELEMENT config (#PCDATA)>">
%config_hex;
''')

with open("templates/index.tpl", "w") as f:
    f.write("""
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <script src="https://cdn.jsdelivr.net/particles.js/2.0.0/particles.min.js"></script>
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;600;800&display=swap');

        :root {
            --text-color: #E0E0E0;
            --highlight: #32a852;
            --shadow: rgba(0, 0, 0, 0.6);
            --navbar-bg: rgba(22, 22, 22, 0.1);
            --border-color: #444;
        }

        * {
            box-sizing: border-box;
            margin: 0;
            padding: 0;
            font-family: 'Inter', sans-serif;
        }

        body {
            background: #0a0a0a;
            color: var(--text-color);
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            height: 100vh;
            padding: 20px;
            position: relative;
        }

        #particles-js {
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            z-index: -1;
        }

        .navbar {
            width: 100%;
            background: var(--navbar-bg);
            display: flex;
            justify-content: space-between;
            align-items: center;
            padding: 20px;
            box-shadow: 0px 4px 15px var(--shadow);
            position: absolute;
            top: 0;
            backdrop-filter: blur(5px);
        }

        .navbar .logo {
            font-size: 24px;
            font-weight: 800;
            color: var(--highlight);
        }

        .navbar .menu {
            display: flex;
            gap: 20px;
            align-items: center;
        }

        .navbar .menu a {
            color: var(--text-color);
            text-decoration: none;
            font-weight: 600;
            padding: 10px 15px;
            transition: 0.3s;
            border-radius: 8px;
            text-align: center;
        }

        .navbar .menu a:hover {
            background: var(--highlight);
            color: #fff;
        }

        .user-info {
            display: flex;
            align-items: center;
            gap: 10px;
        }

        .user-info img {
            width: 40px;
            height: 40px;
            border-radius: 50%;
            border: 2px solid var(--highlight);
            object-fit: cover;
            vertical-align: middle;
        }

        .user-info span {
            font-size: 16px;
            font-weight: 600;
            display: flex;
            align-items: center;
        }

        .content {
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            flex-grow: 1;
            text-align: center;
        }

        h2 {
            color: var(--highlight);
            font-weight: 800;
            margin-bottom: 20px;
        }

        .palette {
            display: flex;
            flex-wrap: wrap;
            justify-content: center;
            gap: 12px;
            padding: 20px;
        }

        .color-box {
            width: 60px;
            height: 60px;
            border-radius: 10px;
            display: flex;
            justify-content: center;
            align-items: center;
            font-size: 14px;
            font-weight: 600;
            color: #FFF;
            text-shadow: 1px 1px 5px rgba(0, 0, 0, 0.5);
            border: 2px solid var(--border-color);
            transition: transform 0.2s ease-in-out;
        }

        .color-box:hover {
            transform: scale(1.1);
            box-shadow: 0px 0px 15px var(--highlight);
        }
    </style>
</head>
<body>
    <div id="particles-js"></div>
    <div class="navbar">
        <div class="logo">Hex Color Palette</div>
        <div class="menu">
            <a href="#">Dashboard</a>
            <a href="#">Settings</a>
            <a href="#">Help</a>
        </div>
        <div class="user-info">
            <img src="https://cdn-yeswehack.com/user/avatar/default_image" style="margin-top: 2px;">
            <span>FullStack HTML Dev</span>
        </div>
    </div>

    <div class="content">
        <h2>Extracted Color Palette</h2>
        <p class="debug">{{ output }}</p>
        <div class="palette">
            {% if colors %}
                {% for color in colors %}
                    <div class="color-box" style="background: {{ color }};">{{ color }}</div>
                {% endfor %}
            {% else %}
                <p class="debug">No colors detected.</p>
            {% endif %}
        </div>
    </div>

    <script>
        particlesJS("particles-js", {
            particles: {
                number: { value: 100 },
                color: { value: "#32a852" },
                shape: { type: "circle" },
                opacity: { value: 0.7 },
                size: { value: 3 },
                move: { speed: 4 },
                line_linked: { enable: true, color: "#32a852" },
            },
            interactivity: {
                detect_on: "canvas",
                events: {
                    onhover: { enable: true, mode: "repulse" },
                    onclick: { enable: true, mode: "push" },
                },
                modes: {
                    repulse: { distance: 100, duration: 0.4 },
                    push: { particles_nb: 4 },
                }
            }
        });
    </script>
</body>
</html>
""")